moonshot/integrations/web_api/services/cookbook_service.py

```diff
@@ -26,40 +26,44 @@ def create_cookbook(self, cookbook_data: CookbookCreateDTO) -> None:
 
     @exception_handler
     def get_all_cookbooks(
-        self, tags: str, categories: str, count: bool
+        self, tags: str, categories: str, count: bool, ids: str | None = None
     ) -> list[CookbookResponseModel]:
         """
         Retrieve all cookbooks, optionally filtered by tags and categories, and with prompt counts.
 
         Args:
+            ids (str): Filter cookbooks by IDs.
             tags (str): Filter cookbooks by tags.
             categories (str): Filter cookbooks by categories.
             count (bool): Include the total prompt count in each cookbook.
 
         Returns:
             list[CookbookResponseModel]: A list of cookbook response models.
         """
-        retn_cookbooks = []
-        cookbooks = moonshot_api.api_get_all_cookbook()
+        cookbooks_list = []
+
+        if ids:
+            cookbook_ids = ids.split(',')
+            cookbooks = [moonshot_api.api_read_cookbook(cookbook_id) for cookbook_id in cookbook_ids]
+        else:
+            cookbooks = moonshot_api.api_get_all_cookbook()
+
+
         for cookbook_dict in cookbooks:
-            cookbook = Cookbook(**cookbook_dict)
-            retn_cookbook = CookbookResponseModel(cookbook=cookbook)
+            cookbook = CookbookResponseModel(**cookbook_dict)
             if count:
-                retn_cookbook.total_prompt_in_cookbook = get_total_prompt_in_cookbook(
+                cookbook.total_prompt_in_cookbook = get_total_prompt_in_cookbook(
                     cookbook
                 )
             if not tags and not categories:
-                retn_cookbooks.append(retn_cookbook)
+                cookbooks_list.append(cookbook)
             if tags and cookbooks_recipe_has_tags(tags, cookbook):
-                retn_cookbooks.append(retn_cookbook)
+                cookbooks_list.append(cookbook)
             if categories and cookbooks_recipe_has_categories(categories, cookbook):
-                if cookbook not in retn_cookbooks:
-                    retn_cookbooks.append(retn_cookbook)
+                if cookbook not in cookbooks_list:
+                    cookbooks_list.append(cookbook)
 
-        return [
-            CookbookResponseModel.model_validate(cookbook)
-            for cookbook in retn_cookbooks
-        ]
+        return cookbooks_list
```

Comment:
@imda-normanchia 
since cookbook is instantiated using CookbookResponseModel, I think we don't have to validate it further here. Can just return the list